src/util/bip32.rs

```diff
@@ -572,7 +644,50 @@ impl ExtendedPubKey {
         })
     }
 
-    /// Returns the HASH160 of the public key of the xpub
+    /// Decoding extended public key from binary data according to BIP 32
+    pub fn decode(data: &[u8]) -> Result<ExtendedPubKey, Error> {
+        if data.len() != 78 {
+            return Err(Error::WrongExtendedKeyLength(data.len()))
+        }
+
+        let cn_int: u32 = endian::slice_to_u32_be(&data[9..13]);
+        let child_number: ChildNumber = ChildNumber::from(cn_int);
+
+        Ok(ExtendedPubKey {
+            network: if data[0..4] == [0x04u8, 0x88, 0xB2, 0x1E] {
+                Network::Bitcoin
+            } else if data[0..4] == [0x04u8, 0x35, 0x87, 0xCF] {
+                Network::Testnet
+            } else {
+                let mut ver = [0u8; 4];
+                ver.copy_from_slice(&data[0..4]);
+                return Err(Error::UnknownVersion(ver));
+            },
+            depth: data[4],
+            parent_fingerprint: Fingerprint::from(&data[5..9]),
+            child_number: child_number,
```

Comment:
In fact this one can be inlined just like the above one to `endian::slice_to_u32_be(&data[9..13]).into()`